modulus/metrics/general/crps.py

```diff
@@ -51,23 +75,41 @@ def kcrps(pred: Tensor, obs: Tensor, dim: int = 0):
     Tensor
         Map of CRPS
     """
-    n = pred.shape[dim]
-    device = pred.device
-    _crps = 0.0 * obs
-    for i in range(n):
-        x_i = torch.index_select(
-            pred, dim, torch.tensor([i], device=device, dtype=torch.int32)
-        )
+    pred = pred.transpose(dim, -1)
+    return _kernel_crps_implementation(pred, obs, unbiased=False)
 
-        x_j = torch.index_select(
-            pred,
-            dim,
-            torch.tensor([j for j in range(i, n)], device=device, dtype=torch.int32),
-        )
 
-        _crps += torch.abs(x_i.squeeze(dim) - obs) / n
-        _crps -= torch.sum(torch.abs(x_i - x_j) / n, dim=dim) / n
-    return _crps
+def fair_crps(pred: Tensor, obs: Union[Tensor, np.ndarray], dim: int = 0) -> Tensor:
+    """Estimate the CRPS from a finite ensemble
+
+        E|X-y|/m - 1/(2m(m-1)) sum_(i,j=1)|x_i - x_j|
+
+    Uses the unbiased estimators described in (Zamo and Naveau, 2018). Unlike
+    ``crps`` this is fair for finite ensembles. Non-fair ``crps`` favors less
+    dispersive ensembles since it is biased high by E|X- X'|/ m where m is the
+    ensemble size.
+
+    Cost: O(m log m) in memory and compute
+
+    Args:
+        pred: (..., m) predictions with ensemble along final dimension
+        obs: (..., ) observations. sharing first dimensions of pred
+
+    Returns
+    -------
+    Tensor
+        Map of CRPS
+
+
+    Notes
+    -----
+    Zamo, M., & Naveau, P. (2018). Estimation of the Continuous Ranked
+    Probability Score with Limited Information and Applications to Ensemble
+    Weather Forecasts. Mathematical Geosciences, 50(2), 209–234.
+    https://doi.org/10.1007/s11004-017-9709-7
+    """
+    pred = pred.transpose(dim, -1)
```

Comment:
This, in addition to similar code above, needs to be replaced with
```
pred = pred.unsqueeze(-1).transpose(dim, -1).squeeze(dim).contiguous()
```

Reply:
good catch. I replaced with [`torch.movedim`](https://pytorch.org/docs/stable/generated/torch.movedim.html#torch.movedim).

Reply:
Great, thanks!